keptn/keptn
# fix(lighthouse-service): Lighthouse now fails if SLI fails  (#6096)

## This PR


1.  lighthouse now fails if SLI fails
2. lighthouse adds a message in case some SLI received were not defined in the SLO
3.  created units tests for these 

### Related Issues


Fixes #6096

### How to test

1. can be tested as follows:

Send an SLI finished event with result fail, see the  output of lighthouse. In my case I changed qualitygate integration test [here](https://github.com/keptn/keptn/blob/master/test/go-tests/qualitygate_test.go#L288) to a ResultFailed and added a message.

![image](https://user-images.githubusercontent.com/89971034/144822838-4df8f887-309b-44d3-b94f-5ba98cbcbd6c.png)

![image](https://user-images.githubusercontent.com/89971034/144822772-0a17d2b1-6393-4c6e-9eca-ec500c0d401a.png)

2. can be tested by adding a metric in an otherwise passing SLI finished event that is not defined in SLO, for instance I changed [here](https://github.com/keptn/keptn/blob/master/test/go-tests/qualitygate_test.go#L307) to a different metric name and run the integration test.
![image](https://user-images.githubusercontent.com/89971034/144824893-fd5743a8-31d6-409f-afff-0a876f6051d5.png)
